backend/src/main/java/org/eventplanner/events/services/ImoListService.java

```diff
@@ -0,0 +1,175 @@
+package org.eventplanner.events.services;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.eventplanner.common.ObjectUtils;
+import org.eventplanner.events.entities.Event;
+import org.eventplanner.events.entities.Registration;
+import org.eventplanner.events.entities.Slot;
+import org.eventplanner.events.values.RegistrationKey;
+import org.eventplanner.positions.adapter.PositionRepository;
+import org.eventplanner.positions.entities.Position;
+import org.eventplanner.positions.values.PositionKey;
+import org.eventplanner.users.entities.UserDetails;
+import org.eventplanner.users.service.UserService;
+import org.eventplanner.users.values.UserKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Service
+public class ImoListService {
+
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    private final UserService userService;
+    private final PositionRepository positionRepository;
+
+    public ImoListService(
+            @Autowired UserService userService,
+            @Autowired PositionRepository positionRepository
+    ) {
+        this.userService = userService;
+        this.positionRepository = positionRepository;
+    }
+
+    public @NonNull ByteArrayOutputStream generateImoList(@NonNull Event event) throws IOException {
+        var positionMap = new HashMap<PositionKey, Position>();
+        positionRepository.findAll().forEach(position -> positionMap.put(position.getKey(), position));
+
+        List<RegistrationKey> assignedRegistrationsKeys = event.getSlots().stream()
+                .map(Slot::getAssignedRegistration)
+                .filter(Objects::nonNull)
+                .toList();
+
+        List<Registration> crewList = event.getRegistrations()
+                .stream()
+                .filter(registration -> assignedRegistrationsKeys.contains(registration.getKey()))
+                .toList();
+
+        FileInputStream fileTemplate = new FileInputStream("data/templates/ImoList_template.xlsx");
+
+        try (fileTemplate) {
```

Comment:
usually the assignment is also within the "try with resource"
See: https://www.baeldung.com/java-try-with-resources